src/ray/raylet/task_dependency_manager.h

```diff
@@ -188,13 +230,16 @@ class TaskDependencyManager {
   /// The storage system for the task lease table.
   gcs::TableInterface<TaskID, TaskLeaseData> &task_lease_table_;
   /// A mapping from task ID of each subscribed task to its list of object
-  /// dependencies.
+  /// dependencies, either task arguments or objects passed into `ray.get`.
   std::unordered_map<ray::TaskID, TaskDependencies> task_dependencies_;
+  /// A mapping from worker ID to each object that the worker called `ray.wait` on.
+  std::unordered_map<ray::WorkerID, WorkerDependencies> worker_dependencies_;
```

Comment:
Instead of keeping both Task and Worker deps, how about keeping only Worker deps and doing the task->worker lookup? Non-actor tasks could clear their worker deps on exit.

Reply:
I don't think this will work out very cleanly because not all tasks have workers (e.g., unassigned tasks that are waiting for object dependencies). I tried something similar before where I only kept track of tasks instead of tasks and workers, but it turns out to be pretty complicated to keep track of the worker->task mapping. For actors, you actually have to track all tasks that still have an active `ray.wait`, even if the tasks already completed.